mutliadmin/MultiAdmin/OutputThread.cs

```diff
@@ -28,362 +28,354 @@ public static ConsoleColor MapConsoleColor(string color, ConsoleColor def = Cons
 
 		public static void Read(Server server)
 		{
-			while (!server.IsStopping())
+			string dedicatedDir = "SCPSL_Data" + Path.DirectorySeparatorChar + "Dedicated";
+			FileSystemWatcher watcher = new FileSystemWatcher();
+			watcher.Path = dedicatedDir;
+			watcher.IncludeSubdirectories = true;
+			watcher.Created += new FileSystemEventHandler((sender, eventArgs) => OnMapiCreated(sender, eventArgs, server));
+			watcher.Filter = "sl*.mapi";
+			watcher.EnableRaisingEvents = true;
+			while (true)
 			{
-				string dedicatedDir = "SCPSL_Data" + Path.DirectorySeparatorChar + "Dedicated" + Path.DirectorySeparatorChar + server.GetSessionId();
-				string[] activeDir = new string[] {};
+				Thread.Sleep(1000);
+			}
+		}
+
+		private static void OnMapiCreated(object source, FileSystemEventArgs e, Server server)
+		{
+			if (!e.FullPath.Contains(server.GetSessionId()))
+			{
+				return;
+			}
+
+			Thread.Sleep(15);
+			OutputThread.ProcessFile(server, e.FullPath);
+		}
+
+		private static void ProcessFile(Server server, string file)
+		{
+			string stream = string.Empty;
+			string command = "open";
+			int attempts = 0;
+			bool read = false;
 
+			while (attempts < (server.runOptimized ? 10 : 100) && !read && !server.IsStopping())
+			{
 				try
 				{
-					if (Directory.Exists(dedicatedDir))
+					if (!File.Exists(file))
 					{
-						activeDir = Directory.GetFiles(dedicatedDir, "sl*.mapi", SearchOption.TopDirectoryOnly).OrderBy(f => f).ToArray<string>();
+						server.Write("Message printer warning: Could not " + command + " " + file + ". File does not exist!", ConsoleColor.Yellow);
+						server.Write("skipping");
+						break;
 					}
+
+					StreamReader sr = new StreamReader(file);
```

Comment:
```suggestion
					using(StreamReader sr = new StreamReader(file))
					{
```

Reply:
This will properly dispose the streamreader

Reply:
Using closure has the same effect as calling `Close()` in the end.

From MSDN
This implementation of Close calls the Dispose method passing a true value.

Reply:
You still need to close the streamreader using this method, but it properly **__disposes__** the streamreader